src/pages/build-apps/guides/data-storage.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Data storage
+description: Save and retrieve data for users with Gaia
+experience: beginners
+tags:
+  - tutorial
+images:
+  large: /images/pages/write-smart-contracts.svg
+  sm: /images/pages/write-smart-contracts-sm.svg
+---
+
+## Introduction
+
+This guide explains how to save and retrieve data for users with [Gaia](/build-apps/references/gaia) by implementing the [`connect`](https://github.com/blockstack/ux/tree/master/packages/connect#stacksconnect) and [`storage`](https://github.com/blockstack/ux/tree/master/packages/storage#stacksstorage) packages of Stacks.js.
+
+Data storage provides a way for users to save both public and private data off-chain while retaining complete control over it.
+
+Storing data off of the blockchain ensures that apps can provide users with high performance and high availability for data reads and writes without the involvement of centralized parties that could comprise their privacy or accessibility.
+
+See [the Todos app tutorial](/build-apps/tutorials/todos) for a concrete example of this functionality in practice.
+
+## Authentication
+
+TODO: Add indicator that authentication guide should be followed first
+
+## How data is stored
+
+Gaia storage is a key-value store.
+
+## Creating a file
+
+Use the [Storage.putFile](https://blockstack.github.io/stacks.js/classes/storage.html#putfile) method:
+
+```tsx
+const userSession = new UserSession();
+const storage = new Storage({ userSession });
+const options: PutFileOptions = {
+  encrypt: false,
+};
+userSession.putFile('hello.txt', 'hello world', options).then(() => {
```

Comment:
Pretty sure the read url of this file is returned here. Would add `const fileUrl = ...` to be more explicit.

Reply:
Cool, added:

![image](https://user-images.githubusercontent.com/28991/104126200-93b25780-535b-11eb-98c6-c50d51fc78ba.png)

![image](https://user-images.githubusercontent.com/28991/104126203-9b71fc00-535b-11eb-9c1e-1816fb5a5c95.png)